File: programmers/스택,큐/python/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    n = len(progresses)
    days = []
    
    for i in range(n):
        pLeft = 100 - progresses[i]
        daysLeft = math.ceil(pLeft / speeds[i])
        days.append(daysLeft)
        
    while days and len(days) != 0:
        count = 1
        d = days.pop(0)
        while d >= days[0]:
            count += 1
            days.pop(0)
            
        answer.append(count)   
         
    return answer

File: programmers/스택,큐/javscript/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let daysLeft = [];
    
    for(let i = 0; i < progresses.length; i++){
        let progLeft = 100 - progresses[i];
        let dayLeft = Math.ceil(progLeft / speeds[i])
        daysLeft.push(dayLeft);
    }
    
    while (daysLeft.length != 0) {
        let j = daysLeft.shift();
        let count = 1;
        while (j >= daysLeft[0]){
            count++;
            daysLeft.shift();
        }
        answer.push(count)
    }
    return answer;
}